### PR TITLE
libsodium: fix MinGW + only raise in validate() + simplify autotools & msvc builds

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

See my reviews in https://github.com/conan-io/conan-center-index/pull/7009

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
test_package still fails if MinGW, I guess due to `-fstack-protector` in `configure.ac`

File: recipes/libsodium/all/conanfile.py [hunk 1]
```diff
@@ -54,89 +61,73 @@ def configure(self):
...
-            host = "{}-w64-mingw32".format("i686" if self.settings.arch == "x86" else self.settings.arch)
-        elif tools.is_apple_os(self.settings.os):
-            host = "{}-apple-{}".format(self.settings.arch, "ios" if self.settings.os == "iOS" else "darwin")
-        elif self.settings.os == "Android":
-            host = "%s-linux-%s" % (tools.to_android_abi(self.settings.arch), self._android_abi_str)
+        if self._is_mingw:
+            self._autotools.libs.append("ssp")
+
+        # TODO: maybe remove this manual host logic? Is it not handled by Autotools helper already?
+        host = None
```
Comment (SSE4) [inline recipes/libsodium/all/conanfile.py:99]: yes, it's old code for QNX Neutrino, which were added before conan learnt to recognize these hosts.
should be no longer needed.

**Review by SSE4 — DISMISSED** [review]

File: recipes/libsodium/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libsodium/all/conanfile.py]: Correct autotools helper relies on `tools.get_gnu_triplet()` which properly handles all these cases. I've pushed a last commit removing all this logic.

**Review by SSE4 — APPROVED** [review]

File: recipes/libsodium/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/libsodium/all/conanfile.py]: okay, less code - less bugs

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**